--- a/videohelpersuite/load_video_nodes.py
+++ b/videohelpersuite/load_video_nodes.py
@@ -170,7 +170,7 @@
 
 def ffmpeg_frame_generator(video, force_rate, frame_load_cap, start_time,
                            custom_width, custom_height, downscale_ratio=8,
-                           meta_batch=None, unique_id=None):
+                           meta_batch=None, unique_id=None, fps_rounding="near"):
     args_input = ["-i", video]
     args_dummy = [ffmpeg_path] + args_input +['-c', 'copy', '-frames:v', '1', "-f", "null", "-"]
     size_base = None
@@ -227,7 +227,7 @@
 
     vfilters = []
     if force_rate != 0:
-        vfilters.append("fps=fps="+str(force_rate))
+        vfilters.append("fps=fps="+str(force_rate)+":round="+fps_rounding)
     if custom_width != 0 or custom_height != 0:
         size = target_size(size_base[0], size_base[1], custom_width,
                            custom_height, downscale_ratio=downscale_ratio)
@@ -248,7 +248,7 @@
     yield (size_base[0], size_base[1], fps_base, duration, fps_base * duration,
            1/(force_rate or fps_base), yieldable_frames, size[0], size[1], alpha)
 
-    args_all_frames += ["-f", "rawvideo", "-"]
+    args_all_frames += ["-fps_mode", "passthrough", "-f", "rawvideo", "-"]
     pbar = ProgressBar(yieldable_frames)
     try:
         with subprocess.Popen(args_all_frames, stdout=subprocess.PIPE) as proc:
@@ -539,6 +539,7 @@
                     "start_time": ("FLOAT", {"default": 0, "min": 0, "max": BIGMAX, "step": .001, "widgetType": "VHSTIMESTAMP"}),
                     },
                 "optional": {
+                    "fps_rounding": (["near", "up", "down"], {"default": "near"}),
                     "meta_batch": ("VHS_BatchManager",),
                     "vae": ("VAE",),
                      "format": get_load_formats(),
